tracer/test/Datadog.Trace.Security.IntegrationTests/Rcm/AspNetCore5AsmAttackerBlocking.cs

```diff
@@ -0,0 +1,139 @@
+        var agent = Fixture.Agent;
+        var result = SubmitRequest(url, null, null, headers: headersAttackerScanner);
+        result.Result.StatusCode.Should().Be(HttpStatusCode.Forbidden);
+
+        var configurationInitial = new[]
+        {
+            ((object)new AppSec.Rcm.Models.Asm.Payload
+            {
+                Actions = new[]
+                {
+                    new Action { Id = "block", Type = BlockingAction.BlockRequestType, Parameters = new AppSec.Rcm.Models.Asm.Parameter { StatusCode = 403, Type = "json" } }
+                },
+            },
+            AsmProduct,
+            nameof(TestSuspiciousAttackerBlocking)),
+            ((object)new AsmFeatures
+            {
+                Asm = new AsmFeature { Enabled = true },
+            },
+            "ASM_FEATURES",
+            nameof(TestSuspiciousAttackerBlocking))
+        };
+
+        await agent.SetupRcmAndWait(Output, configurationInitial);
+        result = SubmitRequest(url, null, null, headers: headersAttackerScanner);
+
+        var exclusions = "[{\"id\": \"exc-000-001\",\"on_match\": \"block_custom\",\"conditions\": [{\"operator\": \"ip_match\",\"parameters\": {\"data\": \"suspicious_ips_data_id\", \"inputs\": [{\"address\": \"http.client_ip\"}]}}]}]";
+        var configuration = new[]
+        {
+            (new AppSec.Rcm.Models.Asm.Payload
+            {
+                Actions = new[]
+                {
+                    new Action { Id = "block_custom", Type = BlockingAction.BlockRequestType, Parameters = new AppSec.Rcm.Models.Asm.Parameter { StatusCode = 405, Type = "auto" } }
+                },
+                Exclusions = (JArray)JToken.Parse(exclusions)
+            },
+            AsmProduct,
+            nameof(TestSuspiciousAttackerBlocking)),
+            ((object)new AppSec.Rcm.Models.AsmData.Payload
+            {
+                ExclusionsData = new[]
+                {
+                    new AppSec.Rcm.Models.AsmData.RuleData { Id = "suspicious_ips_data_id", Type = "ip_with_expiration", Data = new AppSec.Rcm.Models.AsmData.Data[] { new() { Value = "34.65.27.85" } } }
+                },
+            },
+            "ASM_DATA",
+            nameof(TestSuspiciousAttackerBlocking)),
+        };
+
+        await agent.SetupRcmAndWait(Output, configuration);
+        result = SubmitRequest(url + "?a=3", null, null, headers: headersAttackerScanner);
+        result.Result.StatusCode.Should().Be(HttpStatusCode.MethodNotAllowed);
+        result = SubmitRequest(url + "?a=4", null, null, headers: headersRegularScanner);
+        result.Result.StatusCode.Should().Be(HttpStatusCode.Forbidden);
+        result = SubmitRequest(url + "?a=5", null, null, headers: headersAttackerArachni);
+        result.Result.StatusCode.Should().Be(HttpStatusCode.MethodNotAllowed);
+        result = SubmitRequest(url + "?a=6", null, null, headers: headersRegularArachni);
+        result.Result.StatusCode.Should().Be(HttpStatusCode.OK);
```

Comment:
no snapshot verify?

Reply:
For this feature, I though that it would be enough to replicate what the system tests check and was failing regarding status in order to make a more simple test and avoid unnecessary snapshots usage.